thirdparty/fluid/modules/gapi/include/opencv2/gapi/util/variant.hpp

```diff
@@ -523,6 +525,7 @@ namespace util
             return false;
         return (eqs[lhs.index()])(lhs.memory, rhs.memory);
     }
+#pragma GCC diagnostic pop
 
     template<typename... Us> bool operator!=(const variant<Us...> &lhs,
                                              const variant<Us...> &rhs)
```

Comment:
Ubuntu 22.04 build fail, tip: "eqs is used uninitialized".
Actually, I try some way to fix it, but don't work, so I just add this ignored macro to disable check.

Reply:
Modification in a submodule doesn't seem right. When the submodule is updated to another version this fix will likely disappear. One approach is applying a patch before a given submodule is built (patch via cmake commands) or adding some extra flags to this module's cmake configuration to disable some checks.

Reply:
Thanks @tomdol , updated.
I add disable flag to thirdparty/CMakeLists.txt, please review it again.

Reply:
I've double checked how fluid is added to OV's dependencies and it looks like it's not a submodule. Sorry for introducing this confusion but I think we need to go back to the previous version of the fix. This is mostly because switching the warning off in `thirdparty/CMakeLists.txt` will propagate to all OV's dependencies and we don't want that either, it should be switched off only for `fluid`.
Alternatively you could modify this file https://github.com/openvinotoolkit/openvino/blob/master/thirdparty/fluid/modules/gapi/CMakeLists.txt and switch off the warnings here.

Reply:
Yes, it is not a git submodule, but it is submodule from OpenCV source code. So I think maybe we need to push OpenCV to update the responding code, and then OpenVINO also upgrade it again.
BTW, I also try add flag to "https://github.com/openvinotoolkit/openvino/blob/master/thirdparty/fluid/modules/gapi/CMakeLists.txt ", but it doesn't work, I don't know why, maybe https://github.com/openvinotoolkit/openvino/blob/master/thirdparty/CMakeLists.txt#L76  **EXCLUDE_FROM_ALL** arise this problem.

Reply:
So I think current modification and my first update both are OK.
For current modification, it only affect total fluid, and fluid is submodule.
For first,  it only fix build error code, no effect for other code.

So the first update is my recommendation. 
How do you think?
Anyway, the fundamental solution is pushing OpenCV to update their code. This is only our WA, don't block OV upgrade for Ubuntu22.04.

Reply:
If this is a temporary solution until a proper fix is added to OpenCV and OpenCV is upgraded in OV - I'm fine with it. Just please create a jira ticket to perform this update and remove the workaround.

Reply:
Ok, we can use JIRA 86845 to track it.